src/scenic/core/object_types.py

```diff
@@ -3,666 +3,1760 @@
+        contactTolerance (float): The maximum distance can be away from a surface to be considered
+          on the surface. Instances are placed at this distance away from a point when the 
+          ``on`` specifier is used. Default value 0.
+        onDirection (`Vector`): The direction used to determine where to place
+          this `Point` on a regipm, when using the modifying ``on`` specifier.
+          See the :sampref:`on {region}` page for more details. Default value is (0,0,1),
+          directly upwards.
+        visibleDistance (float): Distance used to determine the visible range of this object.
+          Default value 50.
+        rayDensity (float): Determines the number of rays used during visibility checks. This value is the
+          density of rays per degree of visible range in one dimension. The total number of rays 
+          sent will be this value squared per square degree of this object's view angles. Default
+          value 5.
+        mutationScale (float): Overall scale of mutations, as set by the
+          :keyword:`mutate` statement. Default value 0 (mutations disabled).
+        positionStdDev (tuple[float, float, float]): Standard deviation of Gaussian noise 
+          for each dimension (x,y,z) to be added to this object's :prop:`position` 
+          when mutation is enabled with scale 1. Default value (1,1,0), mutating only the x,y values
+          of the point.
+    """
+    _scenic_properties = {
+        "position": PropertyDefault((), {'dynamic'}, lambda self: Vector(0, 0, 0)),
+        "width": 0,
+        "length": 0,
+        "height": 0,
+
+        "baseOffset": Vector(0,0,0),
+        "contactTolerance": 0,
+        'onDirection': Vector(0,0,1),
+
+        "visibleDistance": 50,
+        "rayDensity": 5,
+
+        "mutationScale": 0,
+        "mutator": PropertyDefault({'positionStdDev'}, {'additive'},
+                                lambda self: PositionMutator((self.positionStdDev))),
+        "positionStdDev": (1,1,0),
+
+        # This property is defined in Object, but we provide a default empty value
+        # for Points for implementation convenience.
+        "regionContainedIn": None,
+
+        # These properties are used internally to store entities that must be able to
+        # or must be unable to observe (view) this entity.
+        "_observingEntity": None,
+        "_nonObservingEntity": None,
+    }
+
+    @cached_property
+    def visibleRegion(self):
+        """The :term:`visible region` of this object.
+
+        The visible region of a `Point` is a sphere centered at its ``position`` with
+        radius ``visibleDistance``.
+        """
+        dimensions = (self.visibleDistance, self.visibleDistance, self.visibleDistance)
+        return SpheroidRegion(position=self.position, dimensions=dimensions)
+
+    @cache
+    def canSee(self, other, occludingObjects=list(), debug=False) -> bool:
```

Comment:
The default value for `occludingObjects` is not hashable, so if this method is ever called without giving a value for that parameter we'll get a `TypeError` from `cache`. It would be better to have this method not be memoized, but convert `occludingObjects` to a tuple and then call a memoized helper.

Stepping back, though, are we sure that memoization helps here? Is this function called more than once per sample with a given `self` and `other`?

Reply:
Ah, good catch on the non hashable default. The memoization is helpful for requirements though, some of which call the same visibility check several times.